nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/state/server/ITZooKeeperStateServerTLS.java

```diff
@@ -0,0 +1,415 @@
+import org.apache.nifi.controller.cluster.SecureClientZooKeeperFactory;
+import org.apache.nifi.controller.cluster.ZooKeeperClientConfig;
+import org.apache.nifi.util.NiFiProperties;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.data.Stat;
+import org.apache.zookeeper.server.ServerCnxnFactory;
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.apache.nifi.leader.election.ITSecureClientZooKeeperFactory.createSecureClientProperties;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+// Testing setting up a ZooKeeperStateServer with TLS
+public class ITZooKeeperStateServerTLS {
+    private static final String KEY_STORE = getPath("keystore.jks");
+    private static final String TRUST_STORE = getPath("truststore.jks");
+    private static final String STORE_TYPE = "JKS";
+    private static final String INSECURE_ZOOKEEPER_PROPS = getPath("insecure.zookeeper.properties");
+    private static final String PARTIAL_ZOOKEEPER_PROPS = getPath("partial.zookeeper.properties");
+    private static final String COMPLETE_ZOOKEEPER_PROPS = getPath("complete.zookeeper.properties");
+    private static final String SECURE_ZOOKEEPER_PROPS = getPath("secure.zookeeper.properties");
+    private static final String ZOOKEEPER_PROPERTIES_FILE_KEY = "nifi.state.management.embedded.zookeeper.properties";
+    private static final String ZOOKEEPER_CNXN_FACTORY = "org.apache.zookeeper.server.NettyServerCnxnFactory";
+    private static final String CONNECT_STRING = "localhost:2281";
+    private static final String QUORUM_CONNECT_STRING = "node0.apache.org:2281,node1.apache.org:2281";
+
+    private static final Map<String, String> INSECURE_NIFI_PROPS = new HashMap<String, String>() {{
+        put(ZOOKEEPER_PROPERTIES_FILE_KEY, INSECURE_ZOOKEEPER_PROPS);
+        put(NiFiProperties.WEB_HTTP_HOST, "localhost");
+        put(NiFiProperties.WEB_HTTP_PORT, "8080");
+        put(NiFiProperties.ZOOKEEPER_CLIENT_SECURE, "false");
+    }};
+
+    private static final String TEST_PASSWORD = "passwordpassword";
+
+    private static final Map<String, String> SECURE_NIFI_PROPS = new HashMap<String, String>() {{
+        put(ZOOKEEPER_PROPERTIES_FILE_KEY, SECURE_ZOOKEEPER_PROPS);
+        put("nifi.web.https.port", "8443");
+        put("nifi.security.keystore", KEY_STORE);
+        put("nifi.security.keystoreType", STORE_TYPE);
+        put("nifi.security.keystorePasswd", TEST_PASSWORD);
+        put("nifi.security.truststore", TRUST_STORE);
+        put("nifi.security.truststoreType", STORE_TYPE);
+        put("nifi.security.truststorePasswd", TEST_PASSWORD);
```

Comment:
See comments regarding replacing strings with references to `NiFiProperties` values.